sympy/solvers/ode.py

```diff
@@ -277,10 +277,10 @@
 #: ``best``, and ``all_Integral`` meta-hints should not be included in this
 #: list, but ``_best`` and ``_Integral`` hints should be included.
 allhints = (
-    "separable",
-    "1st_exact",
     "1st_linear",
     "Bernoulli",
+    "separable",
+    "1st_exact",
```

Comment:
@asmeurer 
I have examined one of the test cases to be showing different result(in terms of value, apart from order).
>eq = exp(2*x)*(u.diff(y)) + y*u - u
>sol=pdsolve(eq, hint='1st_linear_variable_coeff')

Without change:
>f(x, y) == exp(y*exp(-2*x))*exp(-y**2*exp(-2*x)/2)*exp(F(x)*exp(-2*x))
![codecogseqn](https://cloud.githubusercontent.com/assets/4049383/5981988/d4a753bc-a8e4-11e4-9d0d-d71b6528b40f.gif)

With change:
>f(x, y) == F(x)*exp(y*exp(-2*x))*exp(-y**2*exp(-2*x)/2)
![codecogseqn 1](https://cloud.githubusercontent.com/assets/4049383/5981990/de4a0ffe-a8e4-11e4-9b34-966f96ae7c7c.gif)